packages/terra-application-utility/src/header/_HeaderUtilityMenu.jsx

```diff
@@ -0,0 +1,256 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import ContentContainer from 'terra-content-container';
+import IconLeft from 'terra-icon/lib/icon/IconLeft';
+import MenuDivider from '../_UtilityMenuDivider';
+import Utils from '../_Utils';
+import MenuItem from './_HeaderUtilityMenuItem';
+import styles from './_HeaderUtilityMenu.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * Key of the top level menu.
+   */
+  initialSelectedKey: PropTypes.string.isRequired,
+  /**
+   * Indicates if the height is bound to a value.
+   */
+  isHeightBounded: PropTypes.bool,
+  /**
+   * The config file containing the menu items to be rendered.
+   */
+  menuItems: PropTypes.arrayOf((Utils.itemShape)).isRequired,
+  /**
+   * The function to trigger when a menu item is selected.
+   */
+  onChange: PropTypes.func.isRequired,
+  /**
+   * The function that closes the menu.
+   */
+  onRequestClose: PropTypes.func,
+};
+
+const defaultProps = {
+  isHeightBounded: undefined,
+};
+
+const processMenuItems = (items) => {
```

Comment:
You're going to want to export this (and hasChevron below) for testing. Or you could define them as statics on the HeaderUtilityMenu class itself.